search_params.go

```diff
@@ -0,0 +1,100 @@
+package stripe
+
+import (
+	"context"
+)
+
+//
+// Public types
+//
+
+// SearchContainer is a general interface for which all search result object structs
+// should comply. They achieve this by embedding a SearchMeta struct and
+// inheriting its implementation of this interface.
+type SearchContainer interface {
+	GetSearchMeta() *SearchMeta
+}
+
+// SearchMeta is the structure that contains the common properties of the search iterators
+type SearchMeta struct {
+	HasMore  bool    `json:"has_more"`
+	NextPage *string `json:"next_page"`
+	URL      string  `json:"url"`
+}
+
+// GetSearchMeta returns a SearchMeta struct (itself). It exists because any
+// structs that embed SearchMeta will inherit it, and thus implement the
+// SearchContainer interface.
+func (l *SearchMeta) GetSearchMeta() *SearchMeta {
+	return l
+}
+
+// SearchParams is the structure that contains the common properties
+// of any *SearchParams structure.
+type SearchParams struct {
+	// Context used for request. It may carry deadlines, cancelation signals,
+	// and other request-scoped values across API boundaries and between
+	// processes.
+	//
+	// Note that a cancelled or timed out context does not provide any
+	// guarantee whether the operation was or was not completed on Stripe's API
+	// servers. For certainty, you must either retry with the same idempotency
+	// key or query the state of the API.
+	Context context.Context `form:"-"`
+
+	Query        string    `form:"query"`
+	Expand       []*string `form:"expand"`
+	SearchWindow *string   `form:"search_window"`
+	SortOrder    *string   `form:"sort_order"`
```

Comment:
We don't have any of these parameters in other languages so I think we can drop them here.